ecosystem/node-checker/src/system_information_evaluator/build_version_evaluator.rs

```diff
@@ -0,0 +1,187 @@
+                BUILD_COMMIT_HASH_KEY
+            ),
+            source: EVALUATOR_SOURCE.to_string(),
+            links: vec![],
+        };
+        get_value(
+            system_information,
+            BUILD_COMMIT_HASH_KEY,
+            evaluation_on_missing_fn,
+        )
+    }
+}
+
+impl SystemInformationEvaluator for BuildVersionEvaluator {
+    /// Assert that the build commit hashes match.
+    fn evaluate_system_information(
+        &self,
+        baseline_system_information: &SystemInformation,
+        target_system_information: &SystemInformation,
+    ) -> Result<Vec<EvaluationResult>, SystemInformationEvaluatorError> {
+        let mut evaluation_results = vec![];
+
+        // Get previous proposals count from the target node.
+        let baseline_build_commit_hash = match self
+            .get_build_commit_hash(baseline_system_information)
+        {
+            GetValueResult::Present(value) => value,
+            GetValueResult::Missing(_evaluation_result) => {
+                return
+                    Err(SystemInformationEvaluatorError::BaselineMissingKey(
+                        BUILD_COMMIT_HASH_KEY.to_string(),
+                        format!("The latest set of metrics from the baseline node did not contain the necessary key \"{}\"", BUILD_COMMIT_HASH_KEY),
+                    ));
+            }
+        };
+
+        let target_build_commit_hash = match self.get_build_commit_hash(target_system_information) {
+            GetValueResult::Present(value) => Some(value),
+            GetValueResult::Missing(evaluation_result) => {
+                evaluation_results.push(evaluation_result);
+                None
+            }
+        };
+
+        match target_build_commit_hash {
+            Some(target_build_commit_hash) => {
+                evaluation_results.push({
+                    if baseline_build_commit_hash == target_build_commit_hash {
+                        EvaluationResult {
+                            headline: "Build commit hashes match".to_string(),
+                            score: 100,
+                            explanation: format!(
+                                "The build commit hash from the target node ({}) matches the build commit hash from the baseline node ({}).",
+                                target_build_commit_hash, baseline_build_commit_hash
+                            ),
+                            source: EVALUATOR_SOURCE.to_string(),
+                            links: vec![],
+                        }
+                    } else {
+                        EvaluationResult {
+                            headline: "Build commit hash mismatch".to_string(),
+                            score: 50,
+                            explanation: format!(
+                                "The build commit hash from the target node ({}) does not match the build commit hash from the baseline node ({}).",
```

Comment:
In some crazy future world, it would be awesome to identify which commit hash is newer 😄